pkg/receive/handler.go

```diff
@@ -193,3 +212,80 @@ func ValidateLabels(logger log.Logger, next http.Handler, labels ...string) http
 		next.ServeHTTP(w, r)
 	}
 }
+
+func (h *Handler) TransformWriteRequest(logger log.Logger, next http.Handler) http.HandlerFunc {
+	logger = log.With(logger, "component", "receive", "middleware", "transformWriteRequest")
```

Comment:
`h.logger` uses `"component", "receive/handler"`, shouldn't this be the same? Another option is using `logger 
 = log.With(h.logger,  "middleware", "transformWriteRequest")`, what do you think?

Reply:
Going with second option!